test/e2e/lib/npd/npd.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package npd
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/node-problem-detector/pkg/util/metrics"
+	"k8s.io/node-problem-detector/test/e2e/lib/gce"
+)
+
+// SetupNPD installs NPD from the test tarball onto the provided GCE instance.
+func SetupNPD(ins gce.Instance, npdBuildTar string) error {
+	tmpDirCmd := ins.RunCommand("mktemp -d")
+	if tmpDirCmd.SSHError != nil || tmpDirCmd.Code != 0 {
+		return fmt.Errorf("error creating temporary directory to hold NPD tarball: %v", tmpDirCmd)
+	}
+
+	tmpDir := strings.TrimSuffix(tmpDirCmd.Stdout, "\n")
+	npdTarVMPath := tmpDir + "/npd.tar.gz"
+	npdExtractDir := tmpDir + "/npd"
+
+	err := ins.PushFile(npdBuildTar, npdTarVMPath)
+	if err != nil {
+		return fmt.Errorf("error pushing local NPD build tarball %s to VM at %s: %v", npdBuildTar, npdTarVMPath, err)
+	}
+
+	mkdirCmd := ins.RunCommand(fmt.Sprintf("mkdir -p %s", npdExtractDir))
+	if mkdirCmd.SSHError != nil || mkdirCmd.Code != 0 {
+		return fmt.Errorf("error creating directory to extract NPD tarball into: %v", mkdirCmd)
+	}
+
+	extractCmd := ins.RunCommand(fmt.Sprintf("tar -xf %s --directory %s", npdTarVMPath, npdExtractDir))
```

Comment:
e2e-install.sh also extracts the tarball, right? Why do we need to extract here?

Reply:
The problem is that the test code need to call `e2e-install.sh` inside the VM. So the first step is, somehow copy `e2e-install.sh` into that VM. I think the simplest implementation is to copy the tarball, then extract the script out of it and call it.

The alternative solution is to separately SCP the `e2e-install.sh` script and the tarball, which I feel is a bit more complex.

WDYT? I don't have a strong preference.

Reply:
SG. Can you add some comment?

Reply:
Done :) Thanks!